rust/yarp/build.rs

```diff
@@ -433,6 +432,13 @@ impl<'pr> Location<'pr> {{
           std::slice::from_raw_parts(start, len)
         }}
     }}
+
+    /// Builds a Location from the given location pointer.
+    #[must_use]
+    pub(crate) fn from_node(node: *mut yp_node_t) -> Location<'pr> {{
```

Comment:
We need a lifetime parameter, because that's how `Location` is defined.  And we have to explicitly use one because -- I think -- Rust can't infer one from an anonymous lifetime parameter in this instance (that's part of the reason we are using/need to use `PhantomData`).  So `'pr` it is.

Reply:
Is rust inferring that the lifetime of the `*mut yp_node_t` here corresponds to the `Location<'pr>`? That was unclear to me.

I'm asking more if you need `from_node<'pr>` here to explicitly accept it?

Reply:
It is not super-obvious to me, either!  We can do `fn from_node<'z>(node: *mut yp_node_t) -> Location<'z>`, which `rustc` is equally happy with, though that feels less clear to me.

My mental model here is that the lifetime matching is being done via the use of `PhantomData`; there's no lifetime information associated with the pointer itself.

Reply:
Let's call in @lopopolo, this seems like something we need clarity on before merging.

Reply:
This is an unbounded lifetime and the compiler can infer it to be anything, including `'static`. This method must take a Rust reference to propagate the lifetime, not a raw pointer.

Reply:
Ahh that matches me intuition. Thanks!

Reply:
@froydnj am I correct in my understanding that you're interested in these bindings for inclusion in Sorbet and the other Rust y'all may be running in production at Stripe?

If that's the case, I'd love to connect with you and your team so we can chat about how I can most effectively help.

My email can be found at my blog.

Reply:
Thanks for the review @lopopolo !  I pushed a new commit that does things without the helper.

I'm interested in eventual yarp integration into Sorbet, but the current goal is playing with the bindings for native extensions.  I keep running into things this crate doesn't do, hence the stream of patches.

Thanks for the offer to chat!  I'll email separately to set up some time.